streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -779,47 +788,64 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
 
     @Override
     public <K1> KGroupedStream<K1, V> groupBy(final KeyValueMapper<? super K, ? super V, K1> selector) {
-        return groupBy(selector, Serialized.with(null, null));
+        return groupBy(selector, Grouped.with(null, null));
     }
 
     @Override
+    @Deprecated
```

Comment:
IMHO, it's better to pass along the deprecation instead of suppressing it.

They both cause the compiler not to issue warnings about the use of deprecated APIs in the method body.

This difference is that if we suppress it here, then any `groupBy` calls on a `KStreamImpl` reference *will not* issue a warning, whereas calls on a `KStream` reference will issue the warning as desired.

Reply:
I don't think that suppress works for any callers of `KStreamImpl#groupBy` -- from my understanding, there will be a warning for all callers independently of a suppress annotation -- callers would need to add their own annotation to suppress the warning for them. A `SuppressWarning` only suppressed warning from the body/implementation of this method (ie, if we would call any other deprecated method).

I also don't think we need `@Deprecated` as this annotation is inherited anyway.

However, this is an internal class anyway, and thus, not public. Thus, I don't have a strong opinion on this.